### PR TITLE
Feature/server metrics

### PR DESCRIPTION
**Pull request by ProofOfKeags**
merge after #401

### CODE REVIEW

**Review by kn0wmad — COMMENTED** [review]
Observing

File: appmgr/src/system.rs [hunk 1]
```diff
@@ -55,11 +56,468 @@ pub async fn logs(
...
+            swap_total,
+            swap_free,
+            swap_used,
+        })
+    }
+}
+
+async fn get_disk_info() -> Result<MetricsDisk, Error> {
+    tokio::task::spawn_blocking(move || {
+        let fs_res = nix::sys::statfs::statfs("/").map_err(|e| Error {
```
Comment (dr-bonez) [inline appmgr/src/system.rs:466]: use Error::new

File: appmgr/src/system.rs [hunk 2]
```diff
@@ -55,11 +56,468 @@ pub async fn logs(
...
+        mem_available: 0,
+        buffers: 0,
+        cached: 0,
+        slab: 0,
+        swap_total: 0,
+        swap_free: 0,
+    };
+    let mut counter = 0;
+    for entry in contents.lines() {
+        if entry.starts_with("MemTotal") {
```
Comment (dr-bonez) [inline appmgr/src/system.rs:405]: use match guards

**Review by dr-bonez — CHANGES_REQUESTED** [review]

File: appmgr/src/system.rs [hunk 3]
```diff
@@ -55,11 +56,468 @@ pub async fn logs(
...
+    general: MetricsGeneral,
+    memory: MetricsMemory,
+    cpu: MetricsCpu,
+    disk: MetricsDisk,
+}
+
+#[command(rpc_only)]
+pub async fn metrics(#[context] ctx: RpcContext) -> Result<Metrics, Error> {
+    let metrics_val = ctx.metrics_cache.read().await;
+    match (*metrics_val).clone() {
```
Comment (dr-bonez) [inline appmgr/src/system.rs:113]:
```suggestion
    match metrics_val.clone() {
```

**Comment by ProofOfKeags** [comment]
Ready for re-review

**Pull request merged**